### PR TITLE
Define preload scope dynamically

### PR DESCRIPTION
**Pull request by dtan4**
## WHY
Now we fix the depth of preload to 2. We would like to fetch no nested fields by default, and specify the depth of preload as we like.

## WHAT
Add new request parameter `preloads` to specify tables to do preload. `fields` parameter is used for choosing fields including in response JSON only.

- If `preloads` parameter is empty, server does not do any preloads.
- `preloads` parameter should be comma-separated string with snake_case table name. Nested tables should be connected with dot. e.g. `profile`, `profile.nation`, `profile.nation,account_name`

Currently we does not specify deeply-nested fields at `fields` parameter (e.g. `profile.nation.name`), but I'll fix this at another branch.

## EXAMPLE
These examples show the behavior with new parameters.

https://gist.github.com/dtan4/97bb2867b869fd9bcda39cbfd3040feb

### CODE REVIEW

File: _templates/controller.go.tmpl [hunk 1]
```diff
@@ -10,33 +11,25 @@ import (
...
+func set{{ .Model.Name }}Preloads(preloads string, db *gorm.DB) *gorm.DB {
+	if preloads == "" {
+		return db
+	}
+
+	for _, preload := range strings.Split(preloads, ",") {
+		var a []string
+
+		for _, s := range strings.Split(preload, ".") {
+			a = append(a, snaker.SnakeToCamel(s))
```
Comment (awakia) [inline _templates/controller.go.tmpl]: Is there any difference from `snaker.SnakeToCamel(preload)`?

Comment (dtan4) [inline _templates/controller.go.tmpl]: For example `profile.working_histories` should be converted as `Profile.WorkingHistories`.
However, do `snaker.SnakeToCamel(preload)` simply results in `Profile.workingHistories`.

Therefore we have to split by dot and do `snaker.SnakeToCamel` for each parts.

Comment (awakia) [inline _templates/controller.go.tmpl]: ok, thanks

File: _templates/controller.go.tmpl [hunk 2]
```diff
@@ -10,33 +11,25 @@ import (
...
-				if v == "{{ if .IsBelongsTo }}{{ .JSONName }}_id{{ else }}id{{ end }}" {
-					idflag = false
-					break
-				}
-			}
-			if idflag {
-				sel = append(sel, "{{ if .IsBelongsTo }}{{ .JSONName }}_id{{ else }}id{{ end }}")
-			}
-{{ end }}{{ end }}
+func set{{ .Model.Name }}Preloads(preloads string, db *gorm.DB) *gorm.DB {
```
Comment (awakia) [inline _templates/controller.go.tmpl]: Currently, this is just same function between models.
So why not rename this to `setPreloads` and put this to appropriate place?

Comment (dtan4) [inline _templates/controller.go.tmpl]: You're right, there is no model-specific code now ❗ I'll rename and move this.

Comment (dtan4) [inline _templates/controller.go.tmpl]: Updated. Move `SetPreloads` into `db/db.go` .

**Comment by awakia** [comment]
Thanks, LGTM!

**Pull request merged**